src/main/java/org/apache/mesos/scheduler/plan/PlanManager.java

```diff
@@ -18,7 +18,14 @@
 
     Optional<Phase> getCurrentPhase();
 
-    Optional<Block> getCurrentBlock();
+    /**
+     * Determines the next block that should be considered for scheduling. Blocks that are being selected by other
+     * PlanManager are provided as {@code dirtiedAssets} as a hint to this PlanManager to assist with scheduling.
+     *
+     * @param dirtiedAssets Other blocks/assets that are already selected for scheduling.
+     * @return An Optional Block that can be scheduled.
```

Comment:
Could specify the situation where the return value may be empty, something like `... that can be scheduled, or an empty Optional when [situation]`